zach_kaz/CounterZK/counterZK.py

```diff
@@ -0,0 +1,24 @@
+from flask import Flask, session, redirect, render_template, request
+app = Flask(__name__)
+app.secret_key = "thisissecret"
+
+@app.route('/')
+def whattodo():
+    if ('x' in session):
+        print('here')
+        session['x']+=1
+    else:
+        session['x']=1
+    return render_template('counterhtml.html',x=session['x'])
```

Comment:
no need to pass the session value as a named argument in render_template. It is also directly available as session['x'] in the html files